packages/integration-sdk-runtime/src/synchronization/index.ts

```diff
@@ -426,6 +440,87 @@ export async function uploadData<T extends UploadDataLookup, K extends keyof T>(
   );
 }
 
+/**
+ * Removes data from the rawData of the largest entity until the overall size
+ * of the data object is less than maxSize (defaulted to UPLOAD_SIZE_MAX).
+ *
+ * @param data
+ */
+export function shrinkRawData<T extends UploadDataLookup, K extends keyof T>(
+  data: T[K][],
+  logger,
+  maxSize = UPLOAD_SIZE_MAX,
+) {
+  const startTimeInMilliseconds = Date.now();
+  let totalSize = Buffer.byteLength(JSON.stringify(data));
+  const initialSize = totalSize;
+  let itemsRemoved = 0;
+  const sizeOfTruncated = Buffer.byteLength("'TRUNCATED'");
+
+  while (totalSize > maxSize) {
+    let largestEntityKey = '';
+    let largestEntitySize = 0;
+    let largestRawDataEntryKey = '';
+    let largestRawDataEntrySize = 0;
+    let largestItemKey = '';
+    let largestItemSize = 0;
+
+    // Find largest Entity
+    for (const entity in data) {
```

Comment:
This is interesting. Is there a reason why you chose to use `in` instead of `of`? 🤔 

I think we should switch this to `of` and just reference the `entity` directly instead of `data.entity`.

Reply:
Fwiw, it might be nice if this was another function too. `largestEntitySize` is _only_ used inside of this block. Would be nice to see something like this:

```ts
const largestEntity = getLargestEntityFromBatch(data);
```